builtin/providers/cloudstack/provider.go

```diff
@@ -1,32 +1,51 @@
 package cloudstack
 
 import (
+	"fmt"
+	"os/user"
+
+	"github.com/go-ini/ini"
 	"github.com/hashicorp/terraform/helper/schema"
 	"github.com/hashicorp/terraform/terraform"
 )
 
 // Provider returns a terraform.ResourceProvider.
 func Provider() terraform.ResourceProvider {
+	usr, err := user.Current()
+	if err != nil {
+		return nil
+	}
```

Comment:
This is still dangerous as the calling side of this function does not expect `nil`, but assumes a `*schema.Provider`. So when you do this and the calling side calls anything on the returned value, it will again panic.